components/victron/victron.cpp

```diff
@@ -8,97 +8,377 @@ namespace victron {
+static const char DEVICE_TYPE_BLUESOLAR_MPPT_100_15_REV2[] PROGMEM = "BlueSolar MPPT 100|15 rev2";
+static const char DEVICE_TYPE_BLUESOLAR_MPPT_75_10_REV3[] PROGMEM = "BlueSolar MPPT 75/10 rev3";
+static const char DEVICE_TYPE_SMARTSOLAR_CHARGER_MPPT_100_30[] PROGMEM = "SmartSolar Charger MPPT 100/30";
+static const char DEVICE_TYPE_SMARTSOLAR_MPPT_RS_450_100[] PROGMEM = "SmartSolar MPPT RS 450/100";
+static const char DEVICE_TYPE_SMARTSOLAR_MPPT_RS_450_200[] PROGMEM = "SmartSolar MPPT RS 450/200";
+
+static const char DEVICE_TYPE_UNKNOWN[] PROGMEM = "Unknown";
+
+// Off reason strings in PROGMEM
 static const uint8_t OFF_REASONS_SIZE = 16;
-static const char *const OFF_REASONS[OFF_REASONS_SIZE] = {
-    "No input power",                       // 0000 0000 0000 0001
-    "Switched off (power switch)",          // 0000 0000 0000 0010
-    "Switched off (device mode register)",  // 0000 0000 0000 0100
-    "Remote input",                         // 0000 0000 0000 1000
-    "Protection active",                    // 0000 0000 0001 0000
-    "Paygo",                                // 0000 0000 0010 0000
-    "BMS",                                  // 0000 0000 0100 0000
-    "Engine shutdown detection",            // 0000 0000 1000 0000
-    "Analysing input voltage",              // 0000 0001 0000 0000
-    "Unknown: Bit 10",                      // 0000 0010 0000 0000
-    "Unknown: Bit 11",                      // 0000 0100 0000 0000
-    "Unknown: Bit 12",                      // 0000 1000 0000 0000
-    "Unknown: Bit 13",                      // 0001 0000 0000 0000
-    "Unknown: Bit 14",                      // 0010 0000 0000 0000
-    "Unknown: Bit 15",                      // 0100 0000 0000 0000
-    "Unknown: Bit 16",                      // 1000 0000 0000 0000
-};
+static const char OFF_REASONS_0[] PROGMEM = "No input power";
+static const char OFF_REASONS_1[] PROGMEM = "Switched off (power switch)";
+static const char OFF_REASONS_2[] PROGMEM = "Switched off (device mode register)";
+static const char OFF_REASONS_3[] PROGMEM = "Remote input";
+static const char OFF_REASONS_4[] PROGMEM = "Protection active";
+static const char OFF_REASONS_5[] PROGMEM = "Paygo";
+static const char OFF_REASONS_6[] PROGMEM = "BMS";
+static const char OFF_REASONS_7[] PROGMEM = "Engine shutdown detection";
+static const char OFF_REASONS_8[] PROGMEM = "Analysing input voltage";
+static const char OFF_REASONS_9[] PROGMEM = "Unknown: Bit 10";
+static const char OFF_REASONS_10[] PROGMEM = "Unknown: Bit 11";
+static const char OFF_REASONS_11[] PROGMEM = "Unknown: Bit 12";
+static const char OFF_REASONS_12[] PROGMEM = "Unknown: Bit 13";
+static const char OFF_REASONS_13[] PROGMEM = "Unknown: Bit 14";
+static const char OFF_REASONS_14[] PROGMEM = "Unknown: Bit 15";
+static const char OFF_REASONS_15[] PROGMEM = "Unknown: Bit 16";
+static const char *const OFF_REASONS[OFF_REASONS_SIZE] PROGMEM = {
```

Comment:
Please don't touch the OFF_REASONS_SIZE/OFF_REASONS. There is no much benefit here.